src/main/java/com/demo/MybatisApplication/service/SubjectService.java

```diff
@@ -20,22 +22,35 @@
 public class SubjectService {
     @Autowired
     SubjectRepository subjectRepository;
-
     @Autowired
     SubjectMapper subjectMapper;
 
-    public SubjectEntityDisplayDto getSubjectById(@PathVariable Long id){
-        return subjectMapper.subjectEntityToSubjectDisplayDto(subjectRepository.findSubjectById(id));
+    public SubjectEntityDisplayDto getSubjectById(@NotNull Long id) {
+        SubjectEntity subjectEntity = subjectRepository.findSubjectById(id);
+        if (subjectEntity == null) {
+            throw new RuntimeException("Subject not found with id: " + id);
+        }
+        return subjectMapper.mapSubjectEntityToSubjectDisplayDto(subjectEntity);
     }
 
     public List<SubjectDisplayDto> getSubjects(){
-        List<SubjectDisplayDto> subjects = subjectMapper.EntitiesToDtos(subjectRepository.findAllSubjects());
+        List<SubjectDisplayDto> subjects = subjectMapper.mapSubjectEntitiesToDtos(subjectRepository.findAllSubjects());
         return subjects;
     }
 
     public List<SubjectAdditionDto> addSubjects(@RequestBody List<SubjectAdditionDto> subjects) {
-        List<SubjectEntity> subjectEntities = subjectMapper.SubjectAddtionDtosToSubjectEntities(subjects);
+        List<SubjectEntity> subjectEntities = subjectMapper.mapSubjectAddtionDtosToSubjectEntities(subjects);
         subjectRepository.saveSubjects(subjectEntities);
         return subjects;
     }
+
+    public ResponseEntity<String> assignSubjectsToStudentHandling(List<Long> subjectIds){
```

Comment:
In this method, we are not assigning Subjects to Students so the name seems to be incorrect. We can call this method as getAllSubjectsByIds. This method will then return the entities for the given Id.
Also, instead of making a database query for each ID we can retrieve all the subject entities in one go for all the given subject Ids.

Reply:
Done